BaseLib/FileTools.cpp

```diff
@@ -37,6 +40,112 @@ bool IsFileExisting(const std::string &strFilename)
     return fs::exists(fs::path(strFilename));
 }
 
+std::tuple<std::string, std::string::size_type, std::string::size_type>
+getParenthesizedString(std::string const& in,
+                       char const open_char,
+                       char const close_char,
+                       std::string::size_type pos)
+{
+    auto const pos_curly_brace_open = in.find_first_of(open_char, pos);
+    if (pos_curly_brace_open == std::string::npos)
+    {
+        return std::make_tuple("", std::string::npos, std::string::npos);
+    }
+    auto const pos_curly_brace_close =
+        in.find_first_of(close_char, pos_curly_brace_open);
+    if (pos_curly_brace_close == std::string::npos)
+    {
+        return std::make_tuple("", std::string::npos, std::string::npos);
+    }
+    return std::make_tuple(
+        in.substr(pos_curly_brace_open + 1,
+                  pos_curly_brace_close - (pos_curly_brace_open + 1)),
+        pos_curly_brace_open, pos_curly_brace_close);
+}
+
+std::string containsKeyword(std::string const& str, std::string const& keyword)
+{
+    auto const position = str.find(keyword);
+    if (position != std::string::npos)
+    {
+        return str.substr(0, position);
+    }
+    return "";
+}
+
+template <typename T>
+bool substituteKeyword(std::string& result, std::string& parenthesized_string,
+                       std::string::size_type begin, std::string::size_type end,
+                       std::string const& keyword, T& data)
+{
+    std::string precision_specification =
+        containsKeyword(parenthesized_string, keyword);
+
+    if (precision_specification.empty())
+    {
+        return false;
+    }
+
+    std::unordered_map<std::type_index, char> type_specification;
```

Comment:
could be static const maybe?

Reply:
Maybe not. Usage is not clear to me. I thought it would be a lookup table. But it is modified later on...

Reply:
I'm not sure if the `unordered_map type_specification` can be `static const` here. It is possible that in different calls of the function the `type_specification` is modified. Furthermore, i wonder if the `static const` would be thread safe?